File: test/fixtures/sample.ts
// Sample TypeScript fixture for AST search testing

interface User {
  id: number;
  name: string;
  email: string;
}

type Status = 'active' | 'inactive' | 'pending';

function createUser(name: string, email: string): User {
  return { id: Date.now(), name, email };
}

const getStatus = (user: User): Status => {
  return user.id > 0 ? 'active' : 'inactive';
};

class UserService {
  private users: User[] = [];

  addUser(user: User): void {
    this.users.push(user);
  }

  findById(id: number): User | undefined {
    return this.users.find(u => u.id === id);
  }
}

export { User, Status, createUser, getStatus, UserService };

File: test/fixtures/sample.py
# Sample Python fixture for AST search testing

import os
from pathlib import Path

def greet(name):
    return f"Hello, {name}!"

def add(a, b):
    return a + b

async def fetch_data(url):
    pass

class Calculator:
    def __init__(self, initial=0):
        self.value = initial

    def add(self, n):
        self.value += n
        return self

    def subtract(self, n):
        self.value -= n
        return self

    def get_result(self):
        return self.value
